### DISCUSSION TITLE
Recently updated git and now I am unable to commit only staged files

### DISCUSSION THREAD

**Discussion by jimifredjr** [community/community#52840] [Programming Help]
### Body

OS: Mac
Git: 2.40.0
I recently upgraded git to 2.40.0 (I don't know the previous version)
and now I'm unable to commit only staged files.

To reproduce:
I have my master branch (master) and a separate branch for feature development (feature)
//Starting in the feature branch
git reset --soft origin/master
//I now have all my staged feature changes
git reset
//My feature changes are unstaged
git add package.json
//I stage 1 of the changed files - package.json
git commit -m 'test commit'
//All my files get added to the commit - I don't think this is the expected behavior
//I would expect only the package.json to be committed

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by airtower-luna** [comment]
That does seem odd. Did you check `git status` between the steps, or can you do that now? If there's really only the one file staged before the commit that could be a bug. :thinking:

**Reply by jimifredjr** [reply]
the comments I put in the post describing where things are staged were checked with git status

**Reply by airtower-luna** [reply]
My point was to double-check that. I agree that with the command sequence only `package.json` should get committed, but if that's not what happens the question is where it goes wrong. :wink:

**Comment by sergeyleschev** [comment]
Another suggestion I would recommend is to try running the git commit command with the -m flag and the file path explicitly specified. For example:
`git commit -m 'commit message' path/to/file.ext`
This should force Git to only commit the explicitly specified file, rather than adding all of the changed files to the commit.

**Reply by jimifredjr** [reply]
I get it but the whole point is I shouldn't have to list the path's to files
That is really cumbersome if I have even 5 files to commit
